examples/tutorials/pipeline_end_to_end_tutorial.ipynb

```diff
@@ -0,0 +1,1472 @@
+      ]
+     },
+     "execution_count": 15,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "# NBVAL_IGNORE_OUTPUT\n",
+    "\n",
+    "# Save the model evaluation result as an artifact\n",
+    "lineapy.save(mod_eval_test, \"iris_model_evaluation\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "278a9931",
+   "metadata": {},
+   "source": [
+    "Note that saving the model evaluation result as an artifact provides advantages in information retrieval. For instance, we can quickly trace the result back to its \"experiment\" setup (i.e., under what conditions the model was trained), like so:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 16,
+   "id": "a23c51cf",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "import pandas as pd\n",
+      "from sklearn.linear_model import LogisticRegression\n",
+      "from sklearn.metrics import classification_report\n",
+      "from sklearn.model_selection import train_test_split\n",
+      "\n",
+      "url = \"https://raw.githubusercontent.com/LineaLabs/lineapy/main/examples/tutorials/data/iris.csv\"\n",
+      "df = pd.read_csv(url)\n",
+      "color_map = {\"Setosa\": \"green\", \"Versicolor\": \"blue\", \"Virginica\": \"red\"}\n",
+      "df[\"variety_color\"] = df[\"variety\"].map(color_map)\n",
+      "num_map = {\"Setosa\": 0, \"Versicolor\": 1, \"Virginica\": 2}\n",
+      "df[\"variety_num\"] = df[\"variety\"].map(num_map)\n",
+      "X = df[[\"petal.length\", \"petal.width\"]]\n",
+      "y = df[\"variety_num\"]\n",
+      "test_size = 0.33\n",
+      "random_state = 42\n",
+      "split_samples = train_test_split(X, y, test_size=test_size, random_state=random_state)\n",
+      "X_train = split_samples[0]\n",
+      "X_test = split_samples[1]\n",
+      "y_train = split_samples[2]\n",
+      "y_test = split_samples[3]\n",
+      "mod = LogisticRegression(multi_class=\"multinomial\")\n",
+      "mod.fit(X_train, y_train)\n",
+      "y_test_pred = mod.predict(X_test)\n",
+      "mod_eval_test = classification_report(y_test, y_test_pred, digits=3)\n",
+      "\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Load the artifact\n",
+    "artifact = lineapy.get(\"iris_model_evaluation\")\n",
+    "\n",
+    "# Check code that generated the artifact\n",
+    "print(artifact.get_code())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "668a82fa",
+   "metadata": {},
+   "source": [
+    "From this, we can quickly know that the result was obtained under training conditions of `test_size = 0.33` and `random_state = 42`, which may help to better compare the result with others as well as devise a new plan for model building. The point is that we are able to know this without separate manual annotation (e.g., \"result from XXX training conditions\") because LineaPy artifacts encapsulate related information altogether."
```

Comment:
This seems like an artifact cool thing ... not related to building pipelines in my opinion. Don't think we need to demo `get_code` and can combine the content of the two cells maybe to say 

"For instance, we can quickly trace the result back to its "experiment" setup (i.e., under what conditions the model was trained), because LineaPy artifacts encapsulate related information altogether.

Reply:
Good point! Yeah, let's help the user stay focused on pipeline functionality.